pkg/controllers/user/logging/clusterlogging.go

```diff
@@ -60,48 +56,50 @@ func registerClusterLogging(cluster *config.UserContext) {
 		rolebindings:         cluster.RBAC.RoleBindings(loggingconfig.LoggingNamespace),
 		clusterRoleBindings:  cluster.RBAC.ClusterRoleBindings(loggingconfig.LoggingNamespace),
 		clusterLister:        cluster.Management.Management.Clusters("").Controller().Lister(),
+		nodeLister:           cluster.Management.Management.Nodes("").Controller().Lister(),
 		clusterName:          cluster.ClusterName,
 	}
-	clusterloggingClient.AddClusterScopedHandler("cluster-logging-controller", cluster.ClusterName, syncer.Sync)
+	clusterloggingClient.AddClusterScopedLifecycle("cluster-logging-controller", cluster.ClusterName, lifecycle)
 }
 
-func (c *ClusterLoggingSyncer) Sync(key string, obj *v3.ClusterLogging) error {
-	//clean up
-	if obj == nil || obj.DeletionTimestamp != nil {
-		return utils.CleanResource(c.namespaces, c.clusterLoggingLister, c.projectLoggingLister)
-	}
+func (c *ClusterLoggingLifecycle) Create(obj *v3.ClusterLogging) (*v3.ClusterLogging, error) {
+	newObj, err := v3.LoggingConditionProvisioned.DoUntilTrue(obj, func() (runtime.Object, error) {
+		return obj, c.provision(obj)
+	})
+	return newObj.(*v3.ClusterLogging), err
```

Comment:
why in Create we don't have to call `createOrUpdateClusterConfigMap ` and create embedded target?
UI will create the CRD first and when user choose any target UI will update the CRD?